insights/parsers/smartctl.py

```diff
@@ -35,12 +35,13 @@ class SMARTctl(Parser):
 
     def __init__(self, context):
         super(SMARTctl, self).__init__(context)
+        self.data = {}
         filename_re = re.compile(r'smartctl_-a_\.dev\.(?P<device>\w+)$')
         match = filename_re.search(context.path)
-        self.data = {}
         if match:
             self.device = match.group('device')
-        # Else warn?  fail?
+        else:
+            self.device = ''
```

Comment:
Also, what would you do with this data if you don't know what device it is related to?